packages/react/src/analytics/integrations/Forter/Forter.ts

```diff
@@ -0,0 +1,192 @@
+    | ((value: SetUserEventData | PromiseLike<SetUserEventData>) => void)
+    | null;
+
+  static override shouldLoad(): boolean {
+    return true;
+  }
+
+  constructor(
+    options: ForterIntegrationOptions,
+    loadData: LoadIntegrationEventData,
+    analytics: StrippedDownAnalytics,
+  ) {
+    const safeOptions = defaultTo({ ...options }, { siteId: '' });
+    super(safeOptions, loadData, analytics);
+
+    if (!safeOptions.siteId) {
+      utils.logger.warn(
+        "[Forter] - `siteId` parameter was not provided in options. It's mandatory to load Forter Integration.",
+      );
+      return;
+    }
+
+    this.initialize();
+  }
+
+  initialize() {
+    const { siteId, origin } = this.options;
+
+    if (!origin) {
+      utils.logger.warn(
+        "[Forter] - `origin` parameter was not provided in options. It's advisable to provide an origin option to aid in debugging",
+      );
+    }
+
+    this.setupOnSetUserPromise();
+
+    this.setupForterTokenLoadedEventListener();
+
+    this.loadForterScript(siteId as string);
+  }
+
+  /**
+   * Sets up a ForterTokenLoadedDetector instance in order to detect
+   * when the forter token loaded event is raised.
+   * When raised, it will then post a tracking message to Omnitracking
+   * with the loaded forter token.
+   *
+   * @returns Promise that will be resolved when the forter token is loaded.
+   */
+  async setupForterTokenLoadedEventListener() {
+    const forterTokenLoadedDetector = new ForterTokenLoadedDetector();
+    const forterToken = await forterTokenLoadedDetector.getForterToken();
+
+    this.sendForterTokenLoadedEventForOmnitracking(forterToken);
+  }
+
+  /**
+   * Creates a synthetic analytics event representing a track of
+   * the forter loaded event in order to be used
+   * in calls to omnitracking-helper functions.
+   *
+   * @returns Promise that will be resolved when analytics creates an event.
+   */
+  async createSyntheticForterTokenLoadedEvent() {
+    const syntheticEventData = await this.strippedDownAnalytics.createEvent(
+      TrackTypes.TRACK,
+      ForterTokenLoadedAnalyticsEvent,
+    );
+
+    return syntheticEventData;
+  }
+
+  /**
+   * Responsible for sending the loaded forter token to Omnitracking.
+   * Needs to wait for a user to be set in analytics by analytics.setUser method
+   * in order to have the `customerId` parameter filled as it is a required
+   * parameter by Omnitracking.
+   *
+   * @param forterToken - String representing the loaded forter token.
+   * @returns Promise that will be resolved when the user is set in analytics and the Omnitracking message is built.
+   *
+   */
+  async sendForterTokenLoadedEventForOmnitracking(forterToken: string) {
+    await this.onSetUserPromise;
+
+    const omnitrackingMessage = await this.getForterTokenLoadedEventPayload(
+      forterToken,
+    );
+
+    if (this.options.omnitrackingHttpClient) {
+      await this.options.omnitrackingHttpClient(omnitrackingMessage);
```

Comment:
I think we should handle failures for this call and the call below from `postTracking` since some browsers might not show a warning if this happens (and in node 16 it will crash the process altogether)